### PR TITLE
Remove default for logit_q parameter to GetQ

### PR DESCRIPTION
**Pull request by Tilps**
Also include a fix draw_score being declared bool in GetBestRootChildWithTemperature.

Basically the same as jio's PR, I just wanted to see what removing the parameter default looked like, so I did it myself...

### CODE REVIEW

File: src/mcts/search.cc [hunk 1]
```diff
@@ -137,7 +137,7 @@ void Search::SendUciInfo() REQUIRES(nodes_mutex_) {
     const auto wl = edge.GetWL();
     const auto d = edge.GetD();
     const int w = static_cast<int>(std::round(500.0 * (1.0 + wl - d)));
-    const auto q = edge.GetQ(default_q, draw_score);
+    const auto q = edge.GetQ(default_q, draw_score, false);
```
Comment (kiudee) [inline src/mcts/search.cc]: Can we add a comment here to make clear what is passed here?
```c++
const auto q = edge.GetQ(default_q, draw_score, /* logit_q= */ false);
```

Comment (Tilps) [reply src/mcts/search.cc]: Done.

Comment (mooskagh) [reply src/mcts/search.cc]: Not really done. :)

Comment (Tilps) [reply src/mcts/search.cc]: Heh I got all the other ones but missed this one? Nice.
Fixed.

**Pull request merged**